Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/JavaFillAlgorithm.kt

```diff
@@ -0,0 +1,204 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.tools.helper
+
+import android.graphics.Bitmap
+import android.graphics.Point
+import androidx.annotation.VisibleForTesting
+import java.util.LinkedList
+import java.util.Queue
+
+private const val UP = true
+private const val DOWN = false
+
+class JavaFillAlgorithm : FillAlgorithm {
+    @VisibleForTesting
+    lateinit var pixels: Array<IntArray>
```

Comment:
Please don't use lateinit, unless it's (like in Activity classes) really necessary.

Reply:
I've used lateinit in the variables here because they are always being set before they are used, and also otherwise a lot of unnecessary null checks would be required throughout the code😅

Reply:
In most cases, one could easily initialize the variables in the constructor, but ok :)